src/windows/main_window.py

```diff
@@ -1874,9 +1874,10 @@ def actionTimelineZoomOut_trigger(self, event):
     def actionFullscreen_trigger(self, event):
         # Toggle fullscreen mode
         if not self.isFullScreen():
+            self.last_window_state = self.windowState()
             self.showFullScreen()
         else:
-            self.showNormal()
+            self.setWindowState(self.last_window_state & ~Qt.WindowFullScreen)
```

Comment:
But that's OK, because the _entire_ `actionFullScreen_trigger()` method can be replaced with [Qt's suggested](https://doc.qt.io/qt-5/qwidget.html#setWindowState) XOR logic:
```python
def actionFullScreen_trigger(self, checked=False):
    # Toggle fullscreen state (current state mask XOR WindowFullScreen)
    self.setWindowState(self.windowState() ^ Qt.WindowFullScreen)
```

Written that way, it really does work as a fullscreen toggle — any other state parameters, like being maximized or snapped to a window border or whatever — are preserved, and only the fullscreen bit is flipped each time.

(I can't submit a proper 'suggestion' if it needs to change any lines outside of the original PR diff; annoying.)